src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -374,7 +374,9 @@ protected function validate(Request $request, Response $entry, $catch = false)
         $subRequest = clone $request;
 
         // send no head requests because we want content
-        $subRequest->setMethod('GET');
+        if (!$request->isMethodSafe() || $request->getMethod() === 'HEAD') {
```

Comment:
I'm pretty sure that `isMethodSafe()` here will always return true. If not, we would not lookup anything in the cache in the first place. So, this part of the condition can be removed.